astrapy/api.py

```diff
@@ -0,0 +1,150 @@
+import logging
+import httpx
+from typing import Any, Dict, Optional, TypeVar, cast
+
+from astrapy.types import API_RESPONSE
+from astrapy.utils import amake_request, make_request
+
+T = TypeVar("T", bound="APIRequestHandler")
+AT = TypeVar("AT", bound="AsyncAPIRequestHandler")
+
+
+logger = logging.getLogger(__name__)
+
+
+class APIRequestError(ValueError):
```

Comment:
I would make `APIRequestError` depend directly on `Exception` as it has different meaning than `ValueError`
```suggestion
class APIRequestError(Exception):
```